File: 2. python/1. basics/lecture 2/p4.py
class Solution:
    def __init__(self, n): #constructor
        for i in range(1,n+1):
            for j in range(1, i+1):
                print(i, end=" ")
            print()

''' pattern output:

1
2 2
3 3 3
4 4 4 4

'''

if __name__ == "__main__":
    size = int(input("enter size: "))

    print("1. output:")
    Solution(size)

File: 1. cpp/1. basics/lecture 2/p4.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
      for(int i=1;i<=n;i++) {
        for(int j=1;j<=i;j++) {
          cout<<i<<" ";
        }
        cout<<endl;
      }
    }
};

/* pattern output:

1
2 2
3 3 3
4 4 4 4

*/

int main() {
  int size;
  cout<<"Enter size of pattern:"<<endl;
  cin>>size;

  cout<<"directly constructor based output:"<<endl;
  solution pattern(size);

}